docs/getting-started/throwing-errors.md

```diff
@@ -0,0 +1,289 @@
+
+
+## Types of error in code
+
+Reliability Kit provides a package to help us differentiate between Operational (known) errors, and Programmer (unknown) errors: [`@dotcom-reliability-kit/errors`](https://github.com/Financial-Times/dotcom-reliability-kit/tree/main/packages/errors#readme).
+
+The `OperationalError` class (explained fully in the [package documentation](https://github.com/Financial-Times/dotcom-reliability-kit/tree/main/packages/errors#operationalerror)) allows you to throw errors which are explicitly marked as "known". Any errors created with this constructor will include an `isOperational` property set to `true`, which can be used by our tooling later to help us understand the health of a system.
+
+```js
+throw new OperationalError({
+    code: 'EXAMPLE_ERROR',
+    message: 'This is an operational error'
+});
+```
+
+This (and further error classes that which extend it) should ideally be used _anywhere_ where you understand how to handle an error. This means in your app you might do something like this:
+
+```js
+// E.g. GET https://your-app/pokemon/oddish
+app.get('/pokemon/:name', async (request, response, next) => {
+    try {
+        // We make a request to an API, we understand that sometimes this API will fail
+        const response = await fetch(
+            `https://pokeapi.co/api/v2/pokemon-species/${request.params.name}`
+        );
+
+        // If the API request failed, we understand that this is an issue with a remote
+        // service which is expected occasionally as part of running our app. Let's throw
+        if (!response.ok) {
+            throw new OperationalError({
+                code: 'POKEMON_API_FAILED',
+                message: `The Pokemon API responded with a ${response.status} status`,
+                relatesToSystems: ['pokeapi']
+            });
+        }
+
+        response.send(await response.json());
+
+    } catch (error) {
+        // Pass any errors onto the Express error handler
+        next(error);
+    }
+});
+```
+
+### Operational errors in library code
+
+Operational errors should not be used in places where an error may be caused by a programmer who's _using_ your library. For example, if you publish the following library to be used by other engineers:
+
+```js
+function uppercaseAll(array) {
+    return array.map(item => item.toUpperCase());
+}
+```
+
+Then you may want to add in some code to check that the data passed in is _definitely_ an array, because otherwise your code won't work correctly. You might add something like this:
+
+```js
+function uppercaseAll(array) {
+    if (!Array.isArray(array)) {
+        throw new OperationalError({
+            message: 'uppercaseAll function expects an array'
+        });
+    }
+    return array.map(item => item.toUpperCase());
+}
+```
```

Comment:
**suggestion**: here, and in a couple of other places, you've introduced an example of what not to do but only explained not to do it _after_ the example. this might be confusing and cause backtracking? would be clearer to explain the examples are bad before the examples.